src/odb.c

```diff
@@ -297,14 +297,9 @@ int git_odb__hashlink(git_oid *out, const char *path)
 		GIT_ERROR_CHECK_ALLOC(link_data);
 
 		read_len = p_readlink(path, link_data, size);
-		link_data[size] = '\0';
-		if (read_len != size) {
-			git_error_set(GIT_ERROR_OS, "failed to read symlink data for '%s'", path);
-			git__free(link_data);
-			return -1;
-		}
```

Comment:
would it be possible to maintain this branch (with the `!= size` turned into `== -1` maybe)? that way the `link_data` does not leak in case of an error.

Reply:
Yep, good call. Fixed.